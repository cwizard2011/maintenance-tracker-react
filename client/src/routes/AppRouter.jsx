import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Home from '../components/Home';
import PrivateRoute from '../utils/PrivateRoute';
import UserDashboard from '../components/UserDashboard';
import AdminDashboard from '../components/AdminDashboard';
import RequestDetails from '../components/RequestDetails';
import NotFoundPage from '../components/NotFoundPage';
import CreateRequest from '../components/CreateRequest';
import EditRequest from '../components/EditRequest';

const AppRouter = () => (
  <BrowserRouter>
    <div>
      <Switch>
        <Route path="/" component={Home} exact />
        <PrivateRoute path="/dashboard" component={UserDashboard} />
        <PrivateRoute path="/request/:requestId" component={RequestDetails} exact />
        <PrivateRoute path="/new-request" component={CreateRequest} />
        <PrivateRoute path="/request/:requestId/edit" component={EditRequest} exact />
        <PrivateRoute path="/admin" component={AdminDashboard} />
        <Route component={NotFoundPage} />
      </Switch>
    </div>
  </BrowserRouter>
);

export default AppRouter;
