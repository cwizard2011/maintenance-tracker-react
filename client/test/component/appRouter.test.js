import React from 'react';
import { shallow } from 'enzyme';
import AppRouter from '../../src/routes/AppRouter';

describe('App Router', () => {
  it('render App correctly', () => {
    const wrapper = shallow(<AppRouter />);
    expect(wrapper).toMatchSnapshot();
  });
});
