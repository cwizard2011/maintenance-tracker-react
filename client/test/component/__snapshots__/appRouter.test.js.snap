// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Router render App correctly 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        path="/dashboard"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        exact={true}
        path="/request/:requestId"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        path="/new-request"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        exact={true}
        path="/request/:requestId/edit"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        path="/admin"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
